develop/tutorials/articles/340-data-upgrades-and-verifiers/creating-a-verify-process-for-your-app.markdown

```diff
@@ -0,0 +1,505 @@
+# Creating A Verify Process for Your App [](id=creating-a-verify-process-for-your-app)
+
+During the cycle of development, you'll want to verify the data your app is 
+producing is consistent and accurate. A verify process is a class that runs 
+when Liferay or your application starts. You use it to verify that your app
+is functioning properly according to its business logic and report any issues
+if it's not. You can do anything from sending a notification of business logic 
+data integrity problems to stopping the server if there's an issue.
+
+Verifiers should only verify that your app is in working order, and they should
+never upgrade the database. Your app's data should only be updated in upgrade 
+processes. You can learn more about upgrade processes in the
+[Creating an Upgrade Process for Your App](/develop/tutorials/-/knowledge_base/7-0/creating-an-upgrade-process-for-your-app) 
+tutorial.
+
+This tutorial covers these topics:
+
+- How to create a verify process for your app using new development patterns
+- How to throw an exception for your verifiers
+- How to execute a verify process automatically and using the Gogo shell
+
+You can get started by creating your verify package next.
+
+## Declaring Verifiers
+
+To create a verify process, you must first create a package for it. Follow these
+steps to create the package and classes for your verify process:
+
+1.  Create a package called `verify` in your project's layout. 
+
+2.  Inside the `verify` package, create an OSGi Component of the service 
+    `VerifyProcess` class that extends the interface `VerifyProcess`:
+
+        @Component(
+                immediate = true,
+                property = {"verify.process.name=
+                com.liferay.verify.process.name"},
+                service = VerifyProcess.class
+        )
+        public class [YourServiceNameVerifyProcess] extends VerifyProcess {
+        
+        ...
+        
+        }
+ 
+    You must use `VerifyProcess.class` as the `service` property value to mark 
+    the class as a valid implementation for the `VerifyProcess` interface. The
+    `immediate` property specifies that the component starts immediately, rather
+    than the first time it's used.
+    
+    You must also define the mandatory `verify.process.name` property, which is 
+    used by the OSGi service tracker for verifiers to capture verifier 
+    components.
```

Comment:
@mdelapenya What does the above sentence mean? Can you break that sentence down into parts so I can understand?
Thanks